starknet/src/interfaces/i_space.cairo

```diff
@@ -151,4 +151,13 @@ trait ISpace<TContractState> {
     fn upgrade(
         ref self: TContractState, class_hash: ClassHash, initialize_calldata: Array<felt252>
     ) -> SyscallResult<()>;
+
+    /// Initializes the contract after an upgrade. This is different from the `initialize` function
+    /// in that it is called after the upgrade, which means the contract already has a state and storage
+    /// associated to it.
+    ///
+    /// # Arguments
+    ///
+    /// * `initialize_calldata` - The calldata to use to perform a post-upgrade initialization.
+    fn post_upgrade_initializer(ref self: TContractState, initialize_calldata: Array<felt252>,);
```

Comment:
do we need this function in the interface for the V1?

Reply:
felt clean to have it even if it doesn't do anything.. makes it clearer for v2 imho but no strong opinion